sample_metadata/parser/generic_metadata_parser.py

```diff
@@ -186,6 +192,28 @@ def get_participant_id(self, row: SingleRow) -> Optional[str]:
             raise ValueError('Participant column does not exist')
         return row[self.participant_column]
 
+    def get_reported_sex(self, row: GroupedRow) -> Optional[int]:
+        """Get reported sex from grouped row"""
+
+        if not self.reported_sex_column:
+            return None
+
+        reported_sex = row[0].get(self.reported_sex_column, None)
+
+        if reported_sex.lower() == 'female':
+            return 2
+        if reported_sex.lower() == 'male':
+            return 1
+        return 0
```

Comment:
I think this should throw if it can't detect, not fall back to undefined - otherwise you might inadvertently lose data.